### PR TITLE
Add typescript, eslint dependencies, types to tests folders

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | The QA Team wants to add typescript to his project
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Tests should run as usual
| Possible impacts? | Automated tests

### CODE REVIEW

**Comment by NeOMakinG** [comment]
@Progi1984 we can't merge it as it is, but I wanted to try to run the CI

We need to remove the 2 any I added and type it properly

**Comment by NeOMakinG** [comment]
@Progi1984 I did type every globals, gave some example using mocha's and playwright's types, now it's your turn!

test:sanity pass

test:all fail because of some wrong code logics using `this`

Also please, provide some more precise types to every `any` used inside globals, I was a bit lazy

File: tests/UI/.eslintrc.js [hunk 1]
```diff
@@ -50,4 +50,27 @@ module.exports = {
     'import/extensions': ['off', 'never'],
     'no-use-before-define': 0,
   },
+  overrides: [
+    {
+      files: ['*.ts'],
+      parser: '@typescript-eslint/parser',
+      plugins: ['@typescript-eslint'],
+      extends: ['prestashop', 'plugin:@typescript-eslint/eslint-recommended', 'plugin:@typescript-eslint/recommended'],
```
Comment (Progi1984) [inline tests/UI/.eslintrc.js]: @NeOMakinG Can I remove `prestashop` as it is not anymore in the `develop` branch ?

**Review by boubkerbribri — DISMISSED** [review]
Same Updated on campaigns and commonTests. All good 
LGTM for the rest.
Did you check all tests using qa.ui.tests.pr ?

**Comment by Progi1984** [comment]
Thanks @boubkerbribri : i have fully forgot. 

That finds some bugs : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/3447716952/jobs/5754019080

PR go in draft ;)

**Review by boubkerbribri — APPROVED** [review]

**Review by nesrineabdmouleh — APPROVED** [review]

**Comment by eternoendless** [comment]
Thank you @NeOMakinG

**Pull request merged**